app/models/post.rb

```diff
@@ -1,3 +1,14 @@
 class Post < ApplicationRecord
-  belongs_to :author_id
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+  before_save :post_count
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def post_count
```

Comment:
- [ ] The same comes here, it would be nice if you save it `after_save` you change rather than `before_save`. Also, do not forget to put your update method after `private`, ok.